api/client_test.go

```diff
@@ -85,7 +85,7 @@ func TestClientDefaultHttpClient_unixSocket(t *testing.T) {
 	if client.addr.Scheme != "http" {
 		t.Fatalf("bad: %s", client.addr.Scheme)
 	}
-	if client.addr.Host != "/var/run/vault.sock" {
+	if client.addr.Host != "localhost" {
```

Comment:
Looking at this more closely, I maybe didn't inspect it enough on the previous PR - this change seems a bit surprising to me, how come we don't have the Unix socket as the host? Maybe it's just a case of the test assertion being out of date, and we need to assert that it's consumed the Unix socket address in a different way?

Reply:
@tomhjp See https://github.com/hashicorp/vault/pull/22523

Reply:
👍 thanks

Reply:
Having the address be `localhost` is the correct behavior. It was a bug to have the client address be set to the socket file path, and the bug was fixed in this PR: https://github.com/hashicorp/vault/pull/22523